### DISCUSSION TITLE
Route for 404?

### DISCUSSION THREAD

**Discussion by ashx404** [vercel/next.js#12957] [Help] [answered]
Is there any way in Next to manually route to 404?
I was trying to working to manually push to 404 to handle a particular case in my application?
I searched in docs, i didn't find anything. 
Currently I am routing it to a page "error404" which is not in my pages folder so it gives a 404. 
I am asking that is there a way to manually redirect it to there?

**Comment by jensmeindertsma** [comment, marked as answer]
Your current approach is probably fine as long as you make sure there is no real page at `/error404`. 

To make sure `error404` doesn't accidentally get used by a real page and cause a problem later on, I would suggest creating a page at `pages/error.js` for example and using that as an error page to which you can route manually like you are already doing.

You could also create a custom error page. To do this, create `404.js` inside your `pages` directory. You can then customize this page. For errors other than a 404, create a `pages/_error.js` page. You can find the documentation for this [in the Next.js documentation](https://nextjs.org/docs/advanced-features/custom-error-page).
